File: javascript/matrix/matrix.js
export class Matrix {
  constructor(matrix_string) {
    this.matrix_string = matrix_string;
  }
  
  get rows() {
    const matrix = this.matrix_string.split("\n")
      .map(str => str.split(' ').map(Number));
    
    Object.defineProperty(this, "rows", { value: matrix });
    return matrix;
  }

  get columns() {
    const mat_transpose = 
      this.rows[0].map((_, i) => this.rows.map(row => row[i]));

    Object.defineProperty(this, "columns", { value: mat_transpose});
    return mat_transpose;
  }
}

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        # Store matrix as 2d integer row-major array
        self.__matrix = [list(map(int, row.split(" "))) for row in matrix_string.split("\n")]

    def row(self, index):
        return self.__matrix[index - 1]

    def column(self, index):
        return list(map(lambda row: row[index - 1], self.__matrix))
